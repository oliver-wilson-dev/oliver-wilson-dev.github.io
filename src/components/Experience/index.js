import React from 'react';
import PropTypes from 'prop-types';
import Section from '../Section';
import Workplace from '../Workplace';


const Experience = ({ title, workplaces, emoji }) => (
  <Section title={title}>
    {workplaces.map(workplace => (
      <Workplace
        key={workplace.companyName}
        {...workplace}
        emoji={emoji}
      />
    ))}
  </Section>
);

Experience.propTypes = {
  emoji: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  workplaces: PropTypes.array.isRequired
};

export default Experience;
