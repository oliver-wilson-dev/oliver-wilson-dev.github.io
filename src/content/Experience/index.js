import React from 'react';
import classnames from 'classnames';
import styles from '../index.css';

export default {
  title: 'Experience',
  content:
  <div>
    <h3 className={styles.companyTitle}>ASOS</h3>
    <h4 className={styles.jobRole}>
      Associate Web Developer
    </h4>
    <h4>September 2017 - present</h4>
    <p>
      <span className={styles.paragraphEmoji} role="img" aria-label="university emoji">ðŸ’»</span>
        After being accepted for the first official graduate scheme for the web platform
        at asos I was catapulted into a number of projects that aided my personal
        and technical growth.
    </p>
    <p>
        During my time at asos I have been tasked with creating an internal react app that
        is used by the photography studios (the team that capture the product images displayed
        on the site) to coordinate their work and allow them to continue to deliver great assets
        while optimising their time and productivity.
    </p>
    <p>
        I have continued to maintain the Product Listing Page (PLP) which is an isomorphic app
        that responds to multiple viewports and is supported by a wide range of browsers.
        SEO is very important for the page as asos needs to stay ahead of its competitors and
        therefore rank higher in search results.
    </p>
    <p>
        I have been a key team member in the replatforming of the Saved Items application,
        employing a tech stack of react, redux and node,
        while using webpack to build the application as well as utilising babel and postcss
        to use the latest syntax available to developers.
    </p>
    <div className={styles.projectImages}>
      <div className={styles.projectImage}>
        <img alt="ASOS Product Listing Page" src="https://github.com/oliver-wilson-dev/oliver-wilson-dev.github.io/blob/master/public/img/plp.png?raw=true" />
      </div>
      <div className={classnames(styles.projectImage, styles.lastProjectImage)}>
        <img alt="ASOS Saved Lists Page" src="https://github.com/oliver-wilson-dev/oliver-wilson-dev.github.io/blob/master/public/img/saved-lists.png?raw=true" />
      </div>
    </div>
  </div>,
};
