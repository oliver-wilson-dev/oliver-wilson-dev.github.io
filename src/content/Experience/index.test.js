import Experience from './index';

describe('Experience Section', () => {
  it('should render correctly', () => {
    expect(Experience).toMatchSnapshot();
  });
});
