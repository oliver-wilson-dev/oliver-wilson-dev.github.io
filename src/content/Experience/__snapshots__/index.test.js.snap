// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience Section should render correctly 1`] = `
Object {
  "content": <div>
    <h3
      className="companyTitle"
    >
      ASOS
    </h3>
    <h4
      className="jobRole"
    >
      Associate Web Developer
    </h4>
    <h4>
      September 2017 - present
    </h4>
    <p>
      <span
        aria-label="university emoji"
        className="paragraphEmoji"
        role="img"
      >
        ðŸ’»
      </span>
      After being accepted for the first official graduate scheme for the web platform at asos I was catapulted into a number of projects that aided my personal and technical growth.
    </p>
    <p>
      During my time at asos I have been tasked with creating an internal react app that is used by the photography studios (the team that capture the product images displayed on the site) to coordinate their work and allow them to continue to deliver great assets while optimising their time and productivity.
    </p>
    <p>
      I have continued to maintain the Product Listing Page (PLP) which is an isomorphic app that responds to multiple viewports and is supported by a wide range of browsers. SEO is very important for the page as asos needs to stay ahead of its competitors and therefore rank higher in search results.
    </p>
    <p>
      I have been a key team member in the replatforming of the Saved Items application, employing a tech stack of react, redux and node, while using webpack to build the application as well as utilising babel and postcss to use the latest syntax available to developers.
    </p>
  </div>,
  "title": "Experience",
}
`;
